crates/fluvio-spu/src/services/public/tests/produce.rs

```diff
@@ -197,3 +197,190 @@ async fn test_produce_invalid_compression() {
+        .expect("filter records");
+
+    let mut produce_request = DefaultProduceRequest {
+        acks: 1,
+        timeout_ms: 300,
+        ..Default::default()
+    };
+
+    let partition_produce = DefaultPartitionRequest {
+        partition_index: 0,
+        records,
+    };
+    let topic_produce_request = TopicProduceData {
+        name: topic.to_owned(),
+        partitions: vec![partition_produce],
+        ..Default::default()
+    };
+
+    produce_request.topics.push(topic_produce_request);
+
+    let produce_response = client_socket
+        .send_and_receive(RequestMessage::new_request(produce_request))
+        .await
+        .expect("send offset");
+
+    // Check error code
+    assert_eq!(produce_response.responses.len(), 1);
+    assert_eq!(produce_response.responses[0].partitions.len(), 1);
+    assert_eq!(
+        produce_response.responses[0].partitions[0].error_code,
+        ErrorCode::None
+    );
+
+    server_end_event.notify();
+    debug!("terminated controller");
+}
+
+#[fluvio_future::test(ignore)]
+async fn test_produce_waiting_replication() {
+    let config = TestConfig::builder()
+        .followers(1_u16)
+        .base_port(14030_u16)
+        .generate("produce_waiting_replication");
+
+    let (leader_ctx, _) = config.leader_replica().await;
+    let public_addr = config.leader_public_addr();
+
+    let public_server_end_event =
+        create_public_server(public_addr.clone(), leader_ctx.clone()).run();
+    let private_server_end_event =
+        create_internal_server(config.leader_addr(), leader_ctx.clone()).run();
+
+    // wait for stream controller async to start
+    sleep(Duration::from_millis(100)).await;
+
+    debug!("starting follower replica controller");
+    config.follower_replica(0).await;
+
+    let client_socket =
+        MultiplexerSocket::new(FluvioSocket::connect(&public_addr).await.expect("connect"));
+    let topic = "test";
+
+    let records_per_request = 5;
+    let records = create_filter_records(records_per_request)
+        .try_into()
+        .expect("filter records");
+
+    let mut produce_request = DefaultProduceRequest {
+        acks: -1,
+        timeout_ms: 10000,
+        ..Default::default()
+    };
+
+    let partition_produce = DefaultPartitionRequest {
+        partition_index: 0,
+        records,
+    };
+    let topic_produce_request = TopicProduceData {
+        name: topic.to_owned(),
+        partitions: vec![partition_produce],
+        ..Default::default()
+    };
+
+    produce_request.topics.push(topic_produce_request);
+
+    let produce_response = client_socket
+        .send_and_receive(RequestMessage::new_request(produce_request))
+        .await
+        .expect("send offset");
+
+    // Check error code
+    assert_eq!(produce_response.responses.len(), 1);
+    assert_eq!(produce_response.responses[0].partitions.len(), 1);
+    assert_eq!(
+        produce_response.responses[0].partitions[0].error_code,
+        ErrorCode::None
```

Comment:
Should check hw of the response to ensure hw is updated

Reply:
There is no hw in the response, only `base_offset`. I added a check from the leader replica state.

Reply:
sounds good